File: 2419-longest-subarray-with-maximum-bitwise-and/2419-longest-subarray-with-maximum-bitwise-and.kt
class Solution {
    fun longestSubarray(nums: IntArray): Int {
        var answer = 0
        var max = 0
        var count = 0
        for (num in nums) {
            if (max < num) {
                max = num
                count = 1
            } else if (max == num) {
                count++
            } else {
                count = 0
            }

            if (answer < count) {
                answer = count
            }
        }

        return answer
    }
}

File: 2419-longest-subarray-with-maximum-bitwise-and/2419-longest-subarray-with-maximum-bitwise-and.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        m = max(nums)
        l = -1
        answer = 0
        for r, num in enumerate(nums):
            if num != m:
                l = r
            else:
                answer = max(answer, r - l)

        return answer